File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.js
function solution(numbers) {
    let arr = [];
    
    // 1. numbers 배열의 number -> String 으로 변환
    numbers.forEach((item) => {
        arr.push(item.toString());
    });
    
    // 2. 정렬 : (b+a) - (a+b)
    arr = arr.sort((a,b) => (b+a) - (a+b)).join('');
    
    return arr[0] === '0' ? '0' : arr;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Collections;

class Solution {
    public String solution(int[] numbers) {
        StringBuilder answer = new StringBuilder();
        
        // 1. numbers 의 숫자들을 문자로 저장
        String[] numbersStr = new String[numbers.length];
        for(int i=0; i<numbers.length; i++) {
            numbersStr[i] = String.valueOf(numbers[i]);
        }
        
        // 2. 문자 내림차순 정렬
        Arrays.sort(numbersStr, (o1,o2) -> (o2+o1).compareTo(o1+o2));
        
        // 3. 내림차순 정렬대로 붙이기
        for(int i=0; i<numbers.length; i++) {
            answer.append(numbersStr[i]);
        }
        
        // 4. 모두 0인 경우
        if(answer.toString().startsWith("0")) {
            return "0";
        }
        
        return answer.toString();
    }
}